ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -251,20 +258,43 @@ protected Node visitJoin(final Join node, final AnalysisContext context) {
     return null;
   }
 
-  private String fetchKeyFieldName(Expression expression) {
+  private Pair<String, String> fetchKeyFieldName(ComparisonExpression comparisonExpression, String sourceAlias, Schema
+      sourceSchema) {
+    Pair<String, String> keyInfo = fetchKeyFieldNameFromExpr(comparisonExpression.getLeft(),
+                                                      sourceAlias, sourceSchema);
+    if (keyInfo == null) {
+      keyInfo = fetchKeyFieldNameFromExpr(comparisonExpression.getRight(), sourceAlias, sourceSchema);
+    }
+    if (keyInfo == null) {
+      throw new KsqlException(String.format("%s : Invalid join criteria %s ", comparisonExpression
+          .getLocation().get().toString(), comparisonExpression));
```

Comment:
calling `get` on an `Opitional` without checking if it is present. Might result in a `NoSuchElementException`

Reply:
Added the check.

Reply:
Did you forget to push the commit? I don't see the check.